multisig-contract/contracts/MultiSig.aes

```diff
@@ -0,0 +1,78 @@
+// ISC License
+//
+// Copyright (c) 2017, aeternity developers
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+// REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+// INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+// LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE
+// OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+// PERFORMANCE OF THIS SOFTWARE.
+
+
+// THIS IS NOT SECURITY AUDITED
+// DO NEVER USE THIS WITHOUT SECURITY AUDIT FIRST
+
+contract MultiSig =
+
+  record state =
+    { admin_one : address
+    , admin_two : address
+    , multi_sig_hashes: map(address, hash) }
+
+  record spend_data =
+    { spend_to_address : address
+    , spend_amount : int }
+
+  public function init(admin_one' : address, admin_two' : address) : state =
+    require_not_zero_address(admin_one')
+    require_not_zero_address(admin_two')
```

Comment:
Why is there a problem if the admin addresses are zero? If you're serious about checking that the admin addresses are correct you should require signatures proving that you have access to the private keys.

Reply:
absolutely right, I did take this from translation of the solidity contract, but will remove it

Reply:
Well, the intent of the non-zero check is just to make sure one did not forget to provide the addresses, simply, as they would else be set to zero @UlfNorell . In order to prove ownership you would of course use signature checks. In our case deployment fails when not providing a value for a param, right? On which level does it do that, call data creation, deployment tx or both @UlfNorell ?

Reply:
Calldata creation fails.